### ISSUE TITLE
Drop support for `Decimal.js`, replace with "Bring your own library" approach

### ISSUE THREAD

**Issue by millsp** [prisma/prisma#21900]
## Breaking change proposal

I propose the following breaking change, to drop support of `Decimal.js` when using `Decimal` and return a `String` instead by default. Optionally, `Decimal.js` or any other alternative can be brought back via the following way:

```ts
import Decimal from 'decimal.js'
new PrismaClient({ decimal: Decimal })
```

In an effort to reduce our bundle size by 32KB, I ask to consider this breaking change. While this is a breaking change, it is also an improvement as it could allow anyone to use their favorite implementation instead with the same philosophy as Driver Adapters (not managed by Prisma, just plugged in).

## Migration path

```ts
import Decimal from 'decimal.js'
new PrismaClient({ decimal: Decimal })
```

This provides an easy and clear way to migrate that needs to be configured only once. The amount of users impacted is probably huge.

**Comment by millsp** [comment]
(Note: Look at prior art)

**Comment by millsp** [comment]
Alternative via client extensions:

```ts
prisma.$extends({
    $serialize: {
        Decimal: {
            output: (value) => new Decimal(value)
            input: (value, default) => {
                if (Decimal.isDecimal(value)) {
                    return value.toFixed()
                }
                return default()
            }
        }
    }
})
```

**Comment by janpio** [comment]
Related: https://github.com/prisma/prisma/issues/13451 + https://github.com/prisma/prisma/issues/7158

**Comment by smaspe** [comment]
+1. would love to avoid importing `Prisma.Decimal` in all things that receive data from Prisma

(note: tried to just import `Decimal` separately, but the 2 are not identical, and so tests tend to fail when mixing them)

**Comment by erikash** [comment]
Would be much better if you’ll return string from Prisma client IMO 🙏🏻

**Comment by bherila** [comment]
Agree. String would be much better. I would prefer to bring my own library, I use currency.js and it's quite inconvenient to translate between that <> Decimal. 
And BTW now with Next 15 + React Server Components, I'd much prefer to pass the string along and only parse it once inside a Client component.

**Comment by alexandre-khoury** [comment]
bump
